### PR TITLE
upgrade/1180-wagtail-django-upgrade

### PR DESCRIPTION
**Pull request by tombola**
https://projects.torchbox.com/projects/rca-django-cms-project/tickets/1180

I upgraded django and psycopg2.

I also went through the other dependencies and updated where I could not see a specific reason not to.

Key upgrades:

- Django: 2.2.19 → 3.1.7
- wagtail: 2.11.2 → 2.12.3
- psycopg2: 2.7.7 → 2.8.6

Updates required by the key ^ upgrades:

- django-redis 4.10.0 → 4.12.1
- redis: 2.10.6 → 3.5.3
- whitenoise 4.1.2 → 5.2.0 (upgrade required for django 3+)
- django-pattern-library 0.2.4 → 0.3.0
- django-storages[boto3] 1.7.1 → 1.11.1 (upgrade required for django 3+)
- django-basic-auth-ip-whitelist: 0.2.1 → 0.3.4

Updates done just to move the project forward:

- django-csp: 3.5 → 3.7
- scout-apm: 2.0.2 → 2.18.0
- wagtail-orderable: 1.0.2 → 1.0.3
- lxml: 4.5.0 → 4.6.3
- django-phonenumber-field: 4.0.0 → 5.0.0
- django-import-export: 2.2.0 → 2.5.0

The sub dependency 'cryptography' caused problems bhy expecting rust to be instaklled during compilation, I pinned it to get round this.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,26 +1,29 @@
-Django==2.2.19
-wagtail==2.11.2
+Django==3.1.7
+wagtail==2.12.3
 # Pin treebeard to <4.5, see: https://github.com/wagtail/wagtail/issues/6820
 django-treebeard<4.5
```
Comment (BrianXu20) [inline requirements.txt:4]: This doc saids that django-treebeard can be unpinned at the version of 

wagtail 2.12.3 

https://docs.wagtail.io/en/stable/releases/2.12.3.html
https://github.com/wagtail/wagtail/commit/de88832899e7a78037776305a271b0d4b2d26ea5

I think the current version will work, and you might wanna try

**Review by BrianXu20 — APPROVED** [review]
LGTM,

File: requirements.txt (hunk 1 above)
Comment (tombola) [reply requirements.txt:4]: Thanks brian, good catch, will test that

**Comment by kevinhowbrook** [comment]
@tombola please do not merge this - it's broken previews on the staging site

**Pull request merged**